File: aat-lib/src/main/java/ch/bailu/aat_lib/map/AppDensity.java
package ch.bailu.aat_lib.map;

public class AppDensity {

        public float toPixel_f(float diPixel) {
            return diPixel;
        }
        public float toPixelScaled_f(float diPixel) {
            return diPixel;
        }


        public int toPixel_i(float diPixel) {
            return (int) (toPixel_f(diPixel)+0.5f);
        }
        public int toPixel_i(float diPixel, int min) {
            return Math.max(min, toPixel_i(diPixel));
        }

        public int toDensityIndependentPixel(float pixel) {
            return (int) (pixel);
        }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/AppDensity.kt
package ch.bailu.aat_lib.map

abstract class AppDensity {
    open fun toPixelFloat(diPixel: Float): Float {
        return diPixel
    }

    open fun toPixelScaledFloat(diPixel: Float): Float {
        return diPixel
    }

    open fun toPixelInt(diPixel: Float): Int {
        return (toPixelFloat(diPixel) + 0.5f).toInt()
    }

    open fun toPixelInt(diPixel: Float, min: Int): Int {
        return Math.max(min, toPixelInt(diPixel))
    }

    open fun toDensityIndependentPixel(pixel: Float): Int {
        return pixel.toInt()
    }
}
